pkg/util/dynamichelper/dynamichelper_test.go

```diff
@@ -4,17 +4,92 @@ package dynamichelper
 // Licensed under the Apache License 2.0.
 
 import (
+	"context"
+	"errors"
 	"reflect"
 	"testing"
 	"time"
 
+	"github.com/golang/mock/gomock"
 	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	kruntime "k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
 
 	"github.com/Azure/ARO-RP/pkg/util/cmp"
+	mock_dynamichelper "github.com/Azure/ARO-RP/pkg/util/mocks/dynamichelper"
 )
 
+func TestEsureDeleted(t *testing.T) {
+	ctx := context.Background()
+
+	for _, tt := range []struct {
+		uname     string
+		grpkind   string
+		namespace string
+		name      string
+		wantErr   error
+		grpvr     *schema.GroupVersionResource
+	}{
+		{
+			uname:     "Test EnsureDeleted 1",
+			grpkind:   "configmap",
+			namespace: "test-namespace-1",
+			name:      "test-name-1",
+			wantErr:   errors.New("this is a new error"),
+			grpvr:     &schema.GroupVersionResource{Group: "", Version: "v1", Resource: "configmaps"},
+		},
+		{
+			uname:     "Test EnsureDeleted 2",
+			grpkind:   "configmap",
+			namespace: "test-namespace-2",
+			name:      "test-name-2",
+			wantErr:   errors.New("This is another error"),
+			grpvr:     &schema.GroupVersionResource{Group: "", Version: "v1", Resource: "configmaps"},
+		},
+	} {
+		t.Run(tt.uname, func(t *testing.T) {
+			mockController := gomock.NewController(t)
+			defer mockController.Finish()
+
+			mockCore := mock_dynamichelper.NewMockGVRResolver(mockController)
+			mockCore.
+				EXPECT().
+				Resolve("configmap", "").
+				Return(nil, tt.wantErr).
```

Comment:
`tt.wantErr` - we usually call the field `wantSomething` in test cases for values we expect. In this case - `tt.wantErr` is mock data because in `Return` we specify what mock is going to return.

---

Unrelated to the above. To set up mock functions we usually have `mocks` function as a field on the test case struct. Like this:

https://github.com/Azure/ARO-RP/blob/d673482fb835b02e4b00dfe0e06135be143c3d07/pkg/api/validate/dynamic/diskencryptionset_test.go#L68-L78

And call it like this:

https://github.com/Azure/ARO-RP/blob/d673482fb835b02e4b00dfe0e06135be143c3d07/pkg/api/validate/dynamic/diskencryptionset_test.go#L274-L276